src/language/concurrency.md

```diff
@@ -235,305 +268,268 @@ After handling the events, the isolate exits.
+  ('vm:isolate-unsendable'), extend or implement such classes cannot be sent
+  through the ports.
+
+Apart from those exceptions any object can be sent. See documentation for
+  [`SendPort`][] for more information.
 
-[`Isolate.exit()`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/Isolate/exit.html
-[`Isolate.spawn()`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/Isolate/spawn.html
-[`ReceivePort`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/ReceivePort-class.html
 [`SendPort`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/SendPort-class.html
 
-You can use these primitives directly for more granular
-control over isolate functionality. For example, `run()` shuts
-down its isolate after returning a single message. 
-What if you want to allow multiple messages to pass between isolates?
-You can set up your own isolate much the same way `run()` is implemented,
-just utilizing the [`send()` method][] of `SendPort` in a slightly different way.
-
-One common pattern, which the following figure shows,
-is for the main isolate to send a request message to the worker isolate,
-which then sends one or more reply messages.
-
-![A figure showing the main isolate spawning the isolate and then sending a request message, which the worker isolate responds to with a reply message; two request-reply cycles are shown](/assets/img/language/concurrency/isolate-custom-bg-worker.png)
-
-Check out the [long_running_isolate.dart][] sample,
-which shows how to spawn a long-running isolate
-that receives and sends messages multiple times between isolates.
-
-{% assign samples = "https://github.com/dart-lang/samples/tree/main/isolates" %}
-
-[isolate samples]: {{ samples }}
-[send_and_receive.dart]: {{ samples }}/bin/send_and_receive.dart
-[long_running_isolate.dart]: {{ samples }}/bin/long_running_isolate.dart
-
-
-## Performance and isolate groups
-
-When an isolate calls [`Isolate.spawn()`][],
-the two isolates have the same executable code
-and are in the same _isolate group_.
-Isolate groups enable performance optimizations such as sharing code;
-a new isolate immediately runs the code owned by the isolate group.
-Also, `Isolate.exit()` works only when the isolates
-are in the same isolate group.
-
-In some special cases,
-you might need to use [`Isolate.spawnUri()`][],
-which sets up the new isolate with a copy of the code
-that's at the specified URI.
-However, `spawnUri()` is much slower than `spawn()`,
-and the new isolate isn't in its spawner's isolate group.
-Another performance consequence is that message passing
-is slower when isolates are in different groups.
-
-[`Isolate.spawnUri()`]: {{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-isolate/Isolate/spawnUri.html
-
-{{site.alert.flutter-note}}
-  Flutter doesn't support `Isolate.spawnUri()`.
-{{site.alert.end}}
-
 <a id="web"></a>
 ## Concurrency on the web
```

Comment:
I know this is another topic the Dart VM team heard user feedback about, wanting more practical examples. If that's covered in the main isolates page, link to it here.

Reply:
We do need to follow up on this, and John has even offered to help on the subject. 

In the Flutter PR about isolates, everyone agreed (including SWEs), that it's outside the scope of this PR, because on the web you don't use the Isolate API.